pkgs/development/libraries/libproxy/default.nix

```diff
@@ -39,22 +42,21 @@ stdenv.mkDerivation rec {
     pcre
     python3
     zlib
+  ] ++ lib.optionals enableJavaScript [
+    jsRuntime
   ] ++ (if stdenv.hostPlatform.isDarwin then [
     SystemConfiguration
     CoreFoundation
-    JavaScriptCore
   ] else [
     glib
-    spidermonkey_68
     dbus
     networkmanager
   ]);
 
   cmakeFlags = [
-    "-DWITH_MOZJS=ON"
     "-DWITH_PYTHON2=OFF"
     "-DPYTHON3_SITEPKG_DIR=${placeholder "py3"}/${python3.sitePackages}"
-  ];
+  ] ++ lib.optional enableJavaScript "-DWITH_MOZJS=ON";
```

Comment:
This is confusing. MozJS refers to SpiderMonkey so it does not make sense that it would enable JavaScriptCore (Webkit).

Reply:
I agree. It's noop on macOS but let's gate it behind `!stdenv.hostPlatform.isDarwin` to make it less confusing.